rules/linux/persistence_systemd_scheduled_timer_created.toml

```diff
@@ -0,0 +1,90 @@
+attackers to regain access in case the connection to the infected asset was lost. 
+"""
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*", "endgame-*"]
+language = "eql"
+license = "Elastic License v2"
+name = "New Systemd Timer Created"
+note = """## Triage and analysis
+
+### Investigating New Systemd Timer Created
+
+Detection alerts from this rule indicate the creation of a new scheduled task through a systemd timer. There are many services that use systemd timers in a benign manner, so investigation to see whether the timer is malicious is vital. The currently set-up systemd timers running as root can be found in the following directories:
+-  /etc/systemd/system/
+-  /usr/local/lib/systemd/system/
+-  /lib/systemd/system/
+-  /usr/lib/systemd/system/
+
+Low privileged users can also set up systemd timers, these can be found in:
+- /home/user/.config/systemd/user/
+
+To list systemd timers, the `systemctl list-timers` command can be leveraged. 
+
+Each systemd timer requires a corresponding systemd service file, with a similar name as the timer. In case the timer was named foo.timer, a foo.service file should be present in the file system. This service file contains the command/script that will be executed by the timer. Investigate the .service file in order to figure out whether the timer is malicious or benign.
+
+### Response and remediation
+
+- Initiate the incident response process based on the outcome of the triage.
+- Isolate the involved host to prevent further post-compromise behavior.
+- Investigate credential exposure on systems compromised or used by the attacker to ensure all compromised accounts are identified. Reset passwords for these accounts and other potentially compromised credentials, such as email, business systems, and web services.
+- Delete the service/timer or restore it to the original configuration.
+- Run a full antimalware scan. This may reveal additional artifacts left in the system, persistence mechanisms, and malware components.
+- Determine the initial vector abused by the attacker and take action to prevent reinfection through the same vector.
+- Leverage the incident response data and logging to improve the mean time to detect (MTTD) and the mean time to respond (MTTR).
+"""
+references = [
+    "https://opensource.com/article/20/7/systemd-timers",
+    "https://pberba.github.io/security/2022/01/30/linux-threat-hunting-for-persistence-systemd-timers-cron/"
+]
+risk_score = 47
+rule_id = "7fb500fa-8e24-4bd1-9480-2a819352602c"
+severity = "medium"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Persistence", "Elastic Endgame", "Investigation Guide"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+file where event.type == "creation" and
+endsWith(file.name, "timer") and
+file.path : (
+  "/etc/systemd/system/*",
+  "/usr/local/lib/systemd/system/*",
+  "/lib/systemd/system/*",
+  "/usr/lib/systemd/system/*",
+  "/home/*/.config/systemd/user/*"
+)
```

Comment:
This is going to turn into whack-a-mole with dev environments and software. How about converting this from an exlusion-by-exception to an inclusion-by-exception rule.

Ex:

```suggestion
) and
(
  /* expand expected bad processes and parent processes */
  process.name in ("bash", "sh", "zsh",) or
  process.parent.name in ()
)
```

Reply:
@brokensound77 But that would cause a similar issue, if not worse, no? As an example, I assume (correct me if I'm wrong) you propose something similar to this:

<img width="1727" alt="image" src="https://user-images.githubusercontent.com/78494512/221529407-ff13b99a-b9f8-47f5-9893-2ad6273d6b0a.png">

There are however so many pre set-up "run-on-boot" or "run-on-set-time-period" processes initiated by systemd that I think logic like the the one proposed in this comment adds more noise than looking for *new* systemd .timer files (as this only occurs when manually creating a new systemd .timer file manually, or installing a service that does this).

Without excluding the ones in my screenshot above, it already triggered 90x on my minimal setup Ubuntu server, and every Linux distro, Python version or Linux Desktop version will have other processes running through systemd.

Additionally, unless we specify every single shell on every linux distro in `cat /etc/shells` (which you could also potentially wildcard by looking for process.name : *sh (to include all dash, bash, rbash, zsh, sh equivalents)) a bypass would still be rather simple by placing any file in any excluded directory (which you can't avoid because of all of the FPs). 

I also thought about sequencing the 2 (looking for the creation of a .timer file, and the execution within a set period of time), but this is also prone to false positives + you can't sequence it effectively if for example an adversary would change a .timer file instead of creating a new one, or executes on reboot which doesn't occur within the same timespan. 

What is your take on this? If I am missing something crucial I'd like to know :) Thanks!

Reply:
I think your original approach is a better, more fine-tuned approach

```sql
file where event.type == "creation" and
endsWith(file.name, "timer") and
file.path : (
  "/etc/systemd/system/*",
  "/usr/local/lib/systemd/system/*",
  "/lib/systemd/system/*",
  "/usr/lib/systemd/system/*",
  "/home/*/.config/systemd/user/*"
)
```

I agree that the `process` version of the rule is even harder to decipher FP vs TP. So the question I have is what would make the timer _creation_ inherently suspicious? Do you think it would be suspicious from certain UIDs or from certain processes. Including those in the original logic scopes this better than adding exclusions for every legitimate use.

I think it is a balance for all rules between exlusion-by-exception and inclusion-by-exception rule. Both have pros and cons. So my question is what logic would you include here that would make it more specific to suspicious timer updates?

Reply:
[This](https://github.com/elastic/detection-rules/pull/2600/files) is an example of whack-a-mole updates for tuning `exclusion-by-exception` rules.

Reply:
Yeah I agree that [This](https://github.com/elastic/detection-rules/pull/2600/files) is not a good base to build upon @brokensound77.

After giving this some more thought, and taking a second look at all of the data that is available within the documents related to this DR, I think the only way to not turn this into whack-a-mole is to do some whitelisting instead of blacklisting. Looking at all of the logs I have available from my different Linux servers; it's the process.name that makes it interesting to analyze or an obvious TN/FP. 

I would propose the following (given the limitations of the data that is available in the documents): the addition of several processes that indicate specific user behavior (manually creating a new timer). This allows a client for easy exception tuning while still maintaining visibility. 

Adding `process.name : ("*sh", "vi*", "nano", "touch", "cp")`. 

Other than this, I thought about making sequencing or process events, but these will all become whack-a-mole. So I think we can either add whitelisting for manual process.names or decide that the benefits of a rule like this do not outweigh the negatives (how easy it would be to by-pass and the unnecessary noise that it generates); implying it might be better to close the PR. 

I do however think that the rule addition mentioned above does have benefits, although relatively easy to bypass.